packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.ts

```diff
@@ -66,15 +80,53 @@ export class ForceSourceRetrieveExecutor extends SfdxCommandletExecutor<
     });
     return quantities;
   }
+
+  public async execute(response: any): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = await new CliCommandExecutor(this.build(response), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    const result = await new CommandOutput().getCmdResult(execution);
+    const resultJson = JSON.parse(result);
```

Comment:
For `JSON.parse` we'd like to have a try-catch block and log telemetry if there's an error.

Reply:
Done!